app/src/org/commcare/activities/FormEntryActivityUIController.java

```diff
@@ -418,7 +418,11 @@ private void showNextView(boolean resuming) {
                             break group_skip;
                         case FormEntryController.EVENT_END_OF_FORM:
                             // auto-advance questions might advance past the last form quesion
-                            refreshCurrentView();
+                            if (questionsView != null) {
```

Comment:
Would be good to add a comment here something like -   `questionsView` null signifies that the form is empty and therefore we would want to auto-save the form.

Reply:
@shubham1g5 added, please check.

Reply:
Being nitpicky here though there is a typo in "special" and don't think the hidden value part is necessary since that's inherent in saving the form.